File: ahammer3-cs50-2018-x-caesar-20181003T112347Z/ahammer3-cs50-2018-x-caesar-20181003T112347Z/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/* This program shifts a string of letters by k letters from plaintext to ciphertext */

int main(int argc, string argv[])
{
    //check if the # arguments is 2
    if (argc != 2)
    {
        printf("Try again - incorrect command line parameters");
        return 1;
    }

    //get key from command line argument, turn into integer
    int k = atoi(argv[1]);

    //prompt user for plaintext
    printf("plaintext: ");
    string plaintext = get_string();

    //for each character in plaintext
    //if alphabetic, preserve case, shift by key and print ciphertext
    printf("ciphertext: ");
    for (int i = 0, n = strlen(plaintext); i < n; i++)
    {
        if (isalpha(plaintext[i]))
        {
            if (isupper(plaintext[i]))
            {
                int shifter = 65;
                printf("%c", (((plaintext[i] + k) - shifter) % 26) + shifter);
            }
            else if (islower(plaintext[i]))
            {
                int shifter = 97;
                printf("%c", (((plaintext[i] + k) - shifter) % 26) + shifter);
            }
        }
        else
        {
            printf("%c", plaintext[i]);
        }
    }
    printf("\n");

    return 0;
}

File: ahammer3-cs50-2018-x-sentimental-caesar-20181107T192639Z/ahammer3-cs50-2018-x-sentimental-caesar-20181107T192639Z/caesar.py
# This program shifts a string of letters by k letters from plaintext to ciphertext

#import modules
import cs50
import sys  # new module sys for python for argv

# account for errors - check if # arguments is 2
if len(sys.argv) != 2:
    print("Try again - incorrect command line parameters")
    sys.exit(1)

# prompt the plaintext and get k from argv
k = int(sys.argv[1])
plaintext = cs50.get_string("Plaintext: ")

# Print the first part of ciphertext
print("ciphertext: ", end="")

# Print the actual ciphertext based on parameters
for char in plaintext:
    if not char.isalpha():
        print(char, end="")
        continue  # Use continue to avoid errors involving non alpha characters

    if char.isupper():
        shifter = 65
    else:
        shifter = 97

    plain = ord(char) - shifter
    cipher = (plain + k) % 26

    print(chr(cipher + shifter), end="")

# Extra endline for aesthetic
print()